### PR TITLE
Global Page: Add Global page filter form and filter logic

### PR DESCRIPTION
**Pull request by ronyyosef**
Added the Global page filter form

### CODE REVIEW

File: static/css/form.css [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  background-color: rgba(0, 128, 0, 0.15);
+  padding: 10px;
+  text-align: center;
+  color: green;
+  border: 1px solid green;
+  border-radius: 3px;
+}
+.submit_container {
+  text-align: center;
+}
```
Comment (adiabramovitch) [inline static/css/form.css:59]: missing a new line

File: static/css/sidebar.css [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  text-align: center;
+  margin-top: 10px;
+}
+
+/* On smaller screens, where height is less than 450px, change the style of the sidenav (less padding and a smaller font size) */
+@media screen and (max-height: 450px) {
+  .sidebar {
+    padding-top: 15px;
+  }
+}
```
Comment (adiabramovitch) [inline static/css/sidebar.css:27]: ditto

File: templates/global/city_dropdown.html [hunk 3]
```diff
@@ -0,0 +1,4 @@
+<option value="">---------</option>
+{% for city in cities %}
+<option value="{{ city.pk }}">{{ city.name }}</option>
+{% endfor %}
```
Comment (adiabramovitch) [inline templates/global/city_dropdown.html:4]: ditto

File: templates/global/filter_form.html [hunk 4]
```diff
@@ -1,3 +1,51 @@
...
+                success: function (data) {
+                    var id = $("#id_city").val();
+                    $("#id_city").html(data);
+                    document.querySelector("#id_city").value = id;
+                },
+            });
+        }
+    </script>
+
+{% endblock %}
```
Comment (adiabramovitch) [inline templates/global/filter_form.html:51]: ditto

File: templates/global/sidebar.html [hunk 5]
```diff
@@ -1,8 +1,9 @@
...
+<link rel="stylesheet" type="text/css" href="{% static 'css/sidebar.css' %}" />
 <body>
   <div>
-    <div class="sidebar">{%include 'global/filter_form.html'%}</div>
+    <div class="sidebar">{% include 'global/filter_form.html'%}</div>
   </div>
 </body>
 
-{% endblock %}
+{% endblock %}
```
Comment (adiabramovitch) [inline templates/global/sidebar.html:9]: ditto

File: house/forms.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+from house.models import House, Job
+
+
+class HouseForm(forms.ModelForm):
+    class Meta:
+        model = House
+        fields = ('country', 'city', 'children')
+
+    jobs = Job.choices.copy()
+    jobs.append((None, '---------'))
```
Comment (adiabramovitch) [inline house/forms.py:13]: Why not add that option to the choises enum?

File: house/forms.py [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    parent_profession_1 = forms.ChoiceField(choices=jobs, required=False)
+    parent_profession_2 = forms.ChoiceField(choices=jobs, required=False)
+    highest_income = forms.IntegerField(label='highest income', required=False)
+    lowest_income = forms.IntegerField(label='lowest income', required=False)
+
```
Comment (adiabramovitch) [inline house/forms.py:18]: I would change the income to be shown like that:
image

File: house/helpers.py [hunk 8]
```diff
@@ -0,0 +1,24 @@
+from django.db.models import QuerySet
+
+from house.models import House
+
+
+def filter_houses_by_form(form_data: dict, filtered_houses: QuerySet[House]) -> QuerySet[House]:
+    for key, value in form_data.items():
+        if value is not None and value != '':
+            if key == 'country':
+                filtered_houses = filtered_houses.filter(country=value)
```
Comment (adiabramovitch) [inline house/helpers.py:10]: I would use match/case it looks a bit cleaner.

**Comment by adiabramovitch** [comment]
Great Job! 
I think that you can add more tests in order to test edge cases for the query. 
It is OK to test things that Django validates just to make sure that the behavior is as expected.

**Comment by adiabramovitch** [comment]
@LiorKGOW @ShellyGolden @Yoavsc129 @orIsrael  @shlior7 
You should review it as well.

File: house/forms.py (hunk 6 above)
Comment (ShellyGolden) [reply house/forms.py:13]: what is the difference between job and  parent_profession_1/parent_profession_1?

File: house/urls.py [hunk 9]
```diff
@@ -9,4 +9,5 @@
     path('login', views.house_login, name='house_login'),
     path('<int:house_id>/', views.house_view, name='house_view'),
     path('house/add', views.add_house, name='add_house'),
+    path('ajax/load-cities/', views.load_cities, name='ajax_load_cities'),
```
Comment (ShellyGolden) [inline house/urls.py:12]: can you explain what is this URL?

File: house/views.py [hunk 10]
```diff
@@ -29,3 +40,10 @@ def house_view(request, house_id):
 
 def add_house(request):
     raise NotImplementedError
+
+
+def load_cities(request):
```
Comment (ShellyGolden) [inline house/views.py:45]: for what we use this function? 
I saw it in the files
Just for understanding

File: templates/global/sidebar.html [hunk 11]
```diff
@@ -1,8 +1,9 @@
 {% block sidebar %}{%load static%}
+<link rel="stylesheet" type="text/css" href="{% static 'css/sidebar.css' %}" />
 <body>
   <div>
-    <div class="sidebar">{%include 'global/filter_form.html'%}</div>
+    <div class="sidebar">{% include 'global/filter_form.html'%}</div>
```
Comment (ShellyGolden) [inline templates/global/sidebar.html:5]: You have unnecessary profit

File: house/tests/house_tests.py [hunk 12]
```diff
@@ -40,14 +42,14 @@ def generate_house(generate_country, generate_city):
...
         assert (
-            generate_house.name == HOUSE_NAME
-            and generate_house.public == HOUSE_PUBLIC
-            and generate_house.country.name == COUNTRY
-            and generate_house.city.name == CITY
-            and generate_house.parent_profession_1 == HOUSE_PARENT_PROFESSION_1
-            and generate_house.parent_profession_2 == HOUSE_PARENT_PROFESSION_2
-            and generate_house.income == HOUSE_INCOME
-            and generate_house.children == HOUSE_CHILDREN
+                generate_house.name == HOUSE_NAME
```
Comment (ShellyGolden) [inline house/tests/house_tests.py:45]: You have unnecessary profits

**Review by ShellyGolden — COMMENTED** [review]
looks great
I left comments

**Comment by ShellyGolden** [comment]
Can you add screenshots? @ronyyosef

**Comment by ronyyosef** [comment]
> 

I

**Comment by ronyyosef** [comment]
@adiabramovitch Can you please tak

File: house/forms.py (hunk 7 above)
Comment (adiabramovitch) [reply house/forms.py:18]: Sure, add it as an issue.

Comment (ronyyosef) [reply house/forms.py:18]: Sure I will add

File: house/tests/house_tests.py [hunk 13]
```diff
@@ -58,3 +61,58 @@ def test_delete_house(self, generate_house):
...
+        House.objects.all().delete()
+
+        # test income filter (lowest_income)
+        pass_filter_house_create()
+        HouseFactory(country=Country.objects.get(name='Israel'), city=City.objects.get(name='Tel Aviv'), income=4000,
+                     children=1, parent_profession_1=Job.TEACHER).save()
+        houses = _filter_houses_by_form(form_data, House.objects.all())
+        assert len(houses) == 1
+        House.objects.all().delete()
+
```
Comment (adiabramovitch) [inline house/tests/house_tests.py:114]: Can you split the test into small functions so each function is a test on its own? 
It is not clear at first read what each test is for. 
As you wrote good code is self-explanatory and comments are not supposed to be critical to the understanding of the reviewer/contributor.
